deployments/cluster/witness.nix

```diff
@@ -1,24 +1,64 @@
-n: { lib, name, nodes, pkgs, ... }: with lib;
+env: n: { lib, name, nodes, pkgs, resources, ... }: with lib;
 
 let
   address = ip: ip + ":4010:4011";
   inherit (nodes.witness-load-balancer.config.networking) publicIPv4;
   isWitness = node: elem "witness" node.config.system.nixos.tags;
+  isInternal = n == 0;
 in
 
 {
-  services.disciplina-witness = {
+  ##
+  # `map` to make additional SGs easier to add and SG list more readable
+  deployment.ec2.securityGroupIds = map (x: resources.ec2SecurityGroups."cluster-${x}-sg".name ) (
+    [ "witness-api-private" ] ++
+      (if isInternal then [ "witness-private" ] else [ "witness-public" ])
+  );
+
+  ## We do not allocate an elastic IP for the internal witness node, so don't try to associate it
+  deployment.ec2.elasticIPv4 = lib.mkIf isInternal (lib.mkForce "");
+
+  ##
+  # For each secret data point:
+  # * Add its file to deployment.keys
+  # * Add its name to services.disciplina-witness.keyFiles
+  # * Wrap its value in `services.disciplina-witness.args` with `cat`
+  deployment.keys = {
+    "witness-comm-sec".keyFile = ../../keys + "/${env}/witness/comm-sec";
```

Comment:
this is the only place `env` is used in this file, maybe it should be somewhere else?